prefect_aws/client_waiter.py

```diff
@@ -0,0 +1,100 @@
+"""Task for waiting on a long-running AWS job"""
+
+import json
+from typing import Any, Dict, Optional
+
+try:
+    import importlib.resources as pkg_resources
+except ImportError:
+    import importlib_resources as pkg_resources
+
+from botocore.waiter import WaiterModel, create_waiter_with_client
+from prefect import get_run_logger, task
+from prefect.utilities.asyncutils import run_sync_in_worker_thread
+
+from prefect_aws import waiters
+from prefect_aws.credentials import AwsCredentials
+
+
+@task
+async def client_waiter(
+    client: str,
+    waiter_name: str,
+    aws_credentials: AwsCredentials,
+    waiter_definition: Optional[Dict[str, Any]] = None,
+    **waiter_kwargs: Optional[Dict[str, Any]],
+):
+    """
+    Uses the underlying boto3 waiter functionality.
+
+    Args:
+        client: The AWS client on which to wait (e.g., 'client_wait', 'ec2', etc).
+        waiter_name: The name of the waiter to instantiate.
+            Can be a boto-supported
+            waiter or one of prefect's custom waiters.
+            Currently, Prefect offers three additional
+            waiters for AWS client_wait:
+
+            - `"JobExists"` waits for a job to be instantiated
+            - `"JobRunning"` waits for a job to start running
+            - `"JobComplete"` waits for a job to finish.
+
+            You can find the definitions for all prefect-defined waiters
+            [here](https://github.com/PrefectHQ/prefect-aws/
+            tree/main/prefect_aws/waiters/).
+            You may also use a custom waiter name, if you supply
+            an accompanying waiter definition dict.
+        credentials: your AWS credentials passed from an upstream
+            Secret task; this Secret must be a JSON string
+            with two keys: `ACCESS_KEY` and `SECRET_ACCESS_KEY` which will be
+            passed directly to `boto3`.  If not provided here or in context, `boto3`
+            will fall back on standard AWS rules for authentication.
+        waiter_definition: A valid custom waiter model, as a dict. Note that if
+            you supply a custom definition, it is assumed that the provided
+            'waiter_name' is contained within the waiter definition dict.
```

Comment:
Maybe include a reference to Prefect's custom waiter definitions as an example of how to do this.